dali/kernels/slice/slice_gpu.cuh

```diff
@@ -99,35 +100,37 @@ union PackedBuffer {
 
 /**
  * @brief Simplified algorithm when no padding is necessary
- * @remarks `in` already refers to the slice anchor start
  */
 template <int Dims, typename OutputType, typename InputType>
 __device__ void SliceFuncNoPad(OutputType *__restrict__ out, const InputType *__restrict__ in,
                                const fast_div<uint64_t> *out_strides, const int64_t *in_strides,
-                               uint64_t offset, uint64_t block_end) {
-  if (Dims > 1 && out_strides[Dims - 1] == static_cast<uint32_t>(in_strides[Dims - 1])) {
+                               const int64_t *anchor, const int64_t *step, uint64_t offset,
+                               uint64_t block_end) {
```

Comment:
Optimization proposal: in absence of padding, we don't really need an anchor or step. We could fuse the step with input strides and calculate them once, before kernel launch. The only change would be that the innermost dimension would have a non-unit input stride.

Reply:
done, passes test_subscript, running operator_2 now
12:30am for me, will continue adding other stuff needed tomorrow